src/engraving/dom/score.cpp

```diff
@@ -5169,14 +5169,28 @@ void Score::changeSelectedNotesVoice(voice_idx_t voice)
         }
 
         if (chord->voice() != voice) {
+            Score* score = this;
+            if (excerpt() && !chord->staff()->isVoiceVisible(voice)) {
+                // We are on a linked stave with the desired voice not visible
+                // Get note in main score to continue
+                note = toNote(note->findLinkedInScore(m_masterScore));
+                chord = note->chord();
+                score = m_masterScore;
+            }
             Segment* s       = chord->segment();
             Measure* m       = s->measure();
             size_t notes     = chord->notes().size();
             track_idx_t dstTrack     = chord->staffIdx() * VOICES + voice;
             ChordRest* dstCR = toChordRest(s->element(dstTrack));
             Chord* dstChord  = nullptr;
 
-            if (excerpt() && muse::key(excerpt()->tracksMapping(), dstTrack, muse::nidx) == muse::nidx) {
+            if (score->excerpt() && muse::key(score->excerpt()->tracksMapping(), dstTrack, muse::nidx) == muse::nidx) {
+                break;
+            }
+
+            if (score->excerpt()
+                && muse::key(score->excerpt()->tracksMapping(), chord->staffIdx() * VOICES + chord->voice(),
+                             muse::nidx) == chord->staffIdx() * VOICES + voice) {
```

Comment:
I _kind_ of know what they mean but these conditions are quite hard to read/understand. Perhaps define a bool variable just before the if to hold the longer part of the statement and give it a name to make it clearer